### ISSUE TITLE
refactor/ rewrite https://github.com/Imgkl/anim_search_bar from scratch

### ISSUE THREAD

**Issue by simon-the-shark** [Solvro/mobile-topwr#163]
I did some patches to make it SOMEHOW work, but's it's still a terrible mess there.

Possibly the best to take the animation (and make it responsive) and rewrite everything else from scratch 
My fork:
https://github.com/simon-the-shark/anim_search_bar
But I literally cried while committing changes there
<img width="1108" alt="Zrzut ekranu 2024-08-17 o 01 02 10" src="https://github.com/user-attachments/assets/5eed3730-891a-4d40-a5e5-12df8d17d556">

**Comment by simon-the-shark** [comment]
ideally then publish on pub.dev

**Comment by simon-the-shark** [comment]
I'm sorry this taking that long :/ I will do it sooner than later

**Comment by wkrzos** [comment]
How's the task going @tomasz-trela, @simon-the-shark?

**Comment by simon-the-shark** [comment]
Yeah @tomasz-trela did smth and i need to check this out and we'll be able to release it soon I guess if we actually start to do things regarding the matter

**Comment by wkrzos** [comment]
The issue doesn't seem all that important for now, so if yo have other things with higher prio, no worries for now.
